server/cluster/unsafe_recovery_controller.go

```diff
@@ -128,9 +128,136 @@ func (u *unsafeRecoveryController) isPlanExecuted(report *pdpb.StoreReport) bool
 	return true
 }
 
+type rangeItem struct {
+	key, value []byte
+}
+
+func (r rangeItem) Less(than btree.Item) bool {
+	return r.key < than.(rangeItem).key
+}
+
 func (u *unsafeRecoveryController) generateRecoveryPlan() {
 	u.Lock()
 	defer u.Unlock()
+	// Peers that cannot elect a leader due to store failures.
+	leaderlessPeers := make([]*pdpb.PeerReport)
+	regionReports := make(map[uint64]*pdpb.PeerReport)
+	for _, storeReport := range u.storeReports {
+		for _, peerReport := range storeReports.Reports {
+			if peerReport.HasLeader {
+				existingReport, ok := regionReports[peerReport.RegionState.Region.Id]
+				if ok && existingReport.RegionState.Region.RegionEpoch.Version > peerReport.RegionState.Region.RegionEpoch.Version {
```

Comment:
compare `ConfVer` too

Reply:
Updated, PTAL. Is it possible that `a.ConfVer > b.ConfVer` while `a.Ver < b.Ver`? If it is,  should we prioritize ConfVer comparison?

Reply:
it's impossible